CONTRIBUTING.md

```diff
@@ -23,3 +23,44 @@
 <em>Refer to the image below for graphically represented Git workflow.</em>
 
 ![Branching strategy](branching.png)
+
+## Pull requests process
+
+When submitting a pull request, make sure to provide a meaningful description. Depending on what your update does, don't hesitate to add context, to
+describe the current (incorrect) behavior, to describe what modification your update brings, etc.
+
+If the PR is related to a specific ticket, include a link to the ticket.
+
+### If your pull request is "approved" by the reviewer
+
+A PR will be approved when there are no changes requested, or when there are only minor changes requested (ex: related to code style or comments, code changes that don't affect the general logic, etc.).
+
+The person that opened the PR will then be responsible to:
+1. Update the code as requested, if changes were requested.
+2. Mark as "resolved" all pending discussions.
+3. **_Squash and Merge_** the PR.
+4. Delete the branch, if it's a _feature_ or _fix_ branch.
```

Comment:
@xfrenette-crim reading that point suggests there is a case where the branch doesn't have to be deleted (like `demo-2.0`), you may want to clarify that point.